database/transaction.go

```diff
@@ -340,19 +337,19 @@ func (tx *Transaction) ReIndex(indexName string) error {
 
 // ReIndexAll truncates and recreates all indexes of the database from scratch.
 func (tx *Transaction) ReIndexAll() error {
-	var indexes []string
+	it := tx.indexStore.st.Iterator(engine.IteratorOptions{})
+	defer it.Close()
```

Comment:
This will panic with Badger because only one iterator is allowed at a time. That's why I didn't use a defer here, sorry I should have added a comment.

Reply:
Ouch, thanks, you are right.

```suggestion
	itClose := it.Close
	defer itClose()
```